src/int.cr

```diff
@@ -395,8 +395,49 @@ struct Int
     (self & mask) == mask
   end
 
-  def gcd(other : Int)
-    self == 0 ? other.abs : (other % self).gcd(self)
+  # Returns the greatest common divisor of `self` and `other`. Signed
+  # integers may raise overflow if either has value equal to `MIN` of
+  # its type.
+  #
+  # Implementation heavily inspired by
+  # https://en.wikipedia.org/wiki/Binary_GCD_algorithm#Iterative_version_in_C
```

Comment:
I don't think such implementation details should be in the API docs, but it can still be on the source code below.
It doesn't impact how the user will use the method.

Reply:
Oh, I missed this. Yeah, for some reason I thought that comment was inside the method definition, not part of its documentation. PRs are welcome, thank you!